relayer/relays/beacon/header/header.go

```diff
@@ -237,14 +243,18 @@ func (h *Header) SyncHeaders(ctx context.Context) error {
 	return nil
 }
 
-func (h *Header) syncInterimFinalizedUpdate(ctx context.Context, lastSyncedSlot uint64) error {
-	checkpointSlot := h.protocol.CalculateNextCheckpointSlot(lastSyncedSlot)
+func (h *Header) syncInterimFinalizedUpdate(ctx context.Context, lastSyncedSlot, newCheckpointSlot uint64) error {
+	// Calculate the range that the interim finalized header update may be in
+	minSlot := newCheckpointSlot - h.protocol.SlotsPerHistoricalRoot
+	maxSlot := lastSyncedSlot + h.protocol.SlotsPerHistoricalRoot
 
-	finalizedUpdate, err := h.syncer.GetLatestPossibleFinalizedUpdate(checkpointSlot, lastSyncedSlot)
+	finalizedUpdate, err := h.syncer.GetFinalizedUpdateAtAttestedSlot(minSlot, maxSlot, false)
```

Comment:
What if `newCheckpointSlot` is far more than the `lastSyncedSlot`?

Reply:
Are you asking what if the `lastSyncedSlot ` is perhaps more than 1 sync committee period behind `newCheckpointSlot `? I also thought about this and I think it can't happen, because the sync committee updates are guaranteed to be sequential, right? Meaning, there will at least be one sync committee update per period.

Reply:
Yeah, it's true. Just ignore it.